community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1765,21 +1777,28 @@ def _get_reservation(self, project: str, zone: str, name: str) -> object:
         """See https://cloud.google.com/compute/docs/reference/rest/v1/reservations"""
         return self.compute.reservations().get(
             project=project, zone=zone, reservation=name).execute()
+    
+    @lru_cache()
+    def _get_future_reservation(self, project:str, zone:str, name: str) -> object:
+        """See https://cloud.google.com/compute/docs/reference/rest/v1/futureReservations"""
+        return self.compute.futureReservations().get(project=project, zone=zone, futureReservation=name).execute()
```

Comment:
What happens with FutureReservation that reached its "end of life"?
Will it throw reasonable expectation if reservation doesn't exist? shall we handle it specifically, e.g. return null?

Reply:
The future reservation seems to last quite a long time even after fulfillment (over 4 months), if the reservation doesn't exist it spits out a 400 error, not sure if that's something we even want to handle because user in that scenario would probably want to redeploy anyway, I'd say no.